packages/@aws-cdk/aws-cognito/lib/user-pool-security.ts

```diff
@@ -0,0 +1,247 @@
+   */
+  readonly actions: AccountTakeoverActions;
+
+  /**
+   * The notify configuration used to construct email notifications.
+   * @default - no configured
+   */
+  readonly notifyConfiguration?: NotifyConfiguration;
+}
+
+/**
+ * The action for compromised credentials.
+ * @see https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pool-settings-compromised-credentials.html
+ */
+export enum CompromisedCredentialsEventAction {
+  /** Blocking requires users to choose another password */
+  BLOCK = 'BLOCK',
+  /** all attempted uses of compromised credentials to Amazon CloudWatch */
+  NO_ACTION = 'NO_ACTION',
+}
+
+/**
+ * The actions for compromised credentials.
+ */
+export interface CompromisedCredentialsActions {
+  /**
+   * The action compromised credentials event.
+   */
+  readonly eventAction: CompromisedCredentialsEventAction;
+}
+
+/**
+ * The event type for compromised credentials check.
+ * @see https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pool-settings-compromised-credentials.html
+ */
+export enum CompromisedCredentialsEvent {
+  /** Users sign in */
+  SIGN_IN = 'SIGN_IN',
+  /** Users change password */
+  PASSWORD_CHANGE = 'PASSWORD_CHANGE',
+  /** Users sign up */
+  SIGN_UP = 'SIGN_UP',
+}
+
+/**
+ * The configuration for compromised credentials risk.
+ */
+export interface CompromisedCredentialsRiskConfiguration {
+  /**
+   * The actions for compromised credentials risk.
+   */
+  readonly actions: CompromisedCredentialsActions;
+  /**
+   * The trigger for compromised credentials check.
+   * @default - Not set.
+   */
+  readonly eventFilter?: CompromisedCredentialsEvent[];
+}
+
+/**
+ * The configuration to override the risk decision.
+ */
+export interface RiskExceptionConfiguration {
+  /**
+   * Overrides the risk decision to always block the pre-authentication requests
+   */
+  readonly blockedIpRangeList?: string[];
+  /**
+   * Risk detection is not performed on the IP addresses in the range list
+   */
+  readonly skippedIpRangeList?: string[];
+}
+
+/**
+ * Risk settings for advanced security.
+ */
+export interface UserPoolRiskConfigurationAttachmentOptions {
+  /**
+   * The identifier of the app client to which you want to specify the risk setting.
+   * You can specify a single client or all clients.
+   * @default 'ALL'
+   */
+  readonly clientId?: string;
+  /**
+   * The account takeover risk configuration including the notify configuration and actions.
+   * @default - no configured
+   */
+  readonly accountTakeoverRisk?: AccountTakeoverRiskConfiguration;
+  /**
+   * The compromised credentials risk configuration including the event filter and action.
+   * @default - no configured
+   */
+  readonly compromisedCredentialsRisk?: CompromisedCredentialsRiskConfiguration;
+  /**
+   * The configuration to override the risk decision.
+   * @default - no configured
+   */
+  readonly riskException?: RiskExceptionConfiguration;
```

Comment:
Avoid this additional `RiskExceptionConfiguration` and directly put the two IP lists here?